horovod/keras/callbacks.py

```diff
@@ -145,43 +150,110 @@ def _autodetect_steps_per_epoch(self):
         else:
             raise ValueError('Could not autodetect the number of steps per epoch. '
                              'Please specify the steps_per_epoch parameter to the '
-                             'LearningRateWarmupCallback() or upgrade to the latest '
-                             'version of Keras.')
+                             '%s() or upgrade to the latest version of Keras.'
+                             % self.__class__.__name__)
+
+    def _adjust_learning_rate(self, epoch):
+        old_lr = K.get_value(self.model.optimizer.lr)
+        new_lr = self.initial_lr * self.multiplier(epoch)
+        K.set_value(self.model.optimizer.lr, new_lr)
+
+        if hasattr(self.model.optimizer, 'momentum') and self.momentum_correction:
+            # See the paper cited above for more information about momentum correction.
+            self.restore_momentum = K.get_value(self.model.optimizer.momentum)
+            K.set_value(self.model.optimizer.momentum,
+                        self.restore_momentum * new_lr / old_lr)
+
+    def _restore_momentum_if_needed(self):
+        if self.restore_momentum:
+            K.set_value(self.model.optimizer.momentum, self.restore_momentum)
+            self.restore_momentum = None
 
     def on_train_begin(self, logs=None):
         self.initial_lr = K.get_value(self.model.optimizer.lr)
-        if not self.steps_per_epoch:
+        if not self.staircase and not self.steps_per_epoch:
             self.steps_per_epoch = self._autodetect_steps_per_epoch()
 
     def on_epoch_begin(self, epoch, logs=None):
         self.current_epoch = epoch
 
     def on_batch_begin(self, batch, logs=None):
-        if self.current_epoch > self.warmup_epochs:
-            # Outside of adjustment scope.
+        if (self.current_epoch < self.start_epoch or
+                self.end_epoch is not None and self.current_epoch >= self.end_epoch):
```

Comment:
Pls enclose the and condition with parentheses so that no need to worry about operator precedence